### DISCUSSION TITLE
[Feature] Auth on Healthcheck path?

### DISCUSSION THREAD

**Discussion by crsnt** [Kong/kong#7916] [Ideas and feature requests]
Hi, new Kong user here. I'm wondering if down the track, we can specify some kind of auth for upstream healthcheck for apps that have protected healthcheck endpoints. 

Thanks.

**Comment by mayocream** [comment]
Hi @crsnt,

In my opinion, [K8s pod liveness probe](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#http-probes) and [consul discovery](https://www.consul.io/docs/discovery/checks) support setting custom HTTP headers for health checks.
Kong can also support that feature.

Kong's health check library do support setting HTTP header for probes, ref: https://github.com/Kong/lua-resty-healthcheck/pull/54.
But Kong doesn't support it in its schema yet: https://github.com/Kong/kong/blob/851484390f1b8d7198a2c84233bbded7cb748dcd/kong/db/schema/entities/upstreams.lua#L87-L123 

Can you elaborate on your use cases?
